### PR TITLE
Update the ISO 15118-2 DC demo with -2 message logs

### PR DESCRIPTION
**Pull request by drmrd**
This PR aligns the demo with what we've been showing during industry outreach events. Changes to the ISO 15118-2 DC demo's Node-Red flows result in all ISO 15118-2-related messages being displayed in a table. The actual request and response messages remain EXI-encoded. A V2G decoder (such as [this one](https://github.com/FlUxIuS/V2Gdecoder) based on RISE-V2G) can then be used to decode these messages.

Future improvements could include incorporating a Node-based V2G decoder into the dashboard or to log un-encoded V2G messages more directly. The message types that are published during the demo's 15118-2 request-response message sequence(s) are displayed above the table in real-time, as well.

### CODE REVIEW

**Review by shankari — COMMENTED** [review]
LGTM as a temporary fix. However, note that the config here is one that I copied from `everest-core` to get the demo to work. We should consider submitting the new config to `everest-core` or wherever we end up with config examples.

@hikinggrass, this is a change to the node-red config in the demo repo. The original file is copied over from `everest-core`, where there are a lot of sample configs. I anticipate that the set of sample configs will grow significantly, as we do more end-to-end testing with various configurations.

We might want to consider where they should live in the long term and how to make it easy for others to discover them.

I have also filed an internal issue to track the potential improvements (toggle on/off, toggle encoded/decoded)

**Review by shankari — APPROVED** [review]

File: docker-compose.iso15118-dc.yml [hunk 1]
```diff
@@ -2,14 +2,14 @@ version: "3.6"
 
 services:
   mqtt-server:
-    image: ghcr.io/shankari/everest-demo/mqtt-server:0.0.1
+    image: ghcr.io/everest/everest-demo/mqtt-server:0.0.1
```
Comment (shankari) [inline docker-compose.iso15118-dc.yml:5]:
```suggestion
    image: ghcr.io/us-joet/everest-demo/mqtt-server:build_pkg_fix_dc_config
```

File: docker-compose.iso15118-dc.yml [hunk 2]
```diff
@@ -2,14 +2,14 @@ version: "3.6"
...
     logging:
       driver: none
     ports:
       - 1883:1883
 
   manager:
-    image: ghcr.io/shankari/everest-demo/manager:0.0.2
+    image: ghcr.io/everest/everest-demo/manager:0.0.2
```
Comment (shankari) [inline docker-compose.iso15118-dc.yml:12]:
```suggestion
    image: ghcr.io/us-joet/everest-demo/manager:build_pkg_fix_dc_config
```

File: docker-compose.iso15118-dc.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ services:
       - net.ipv6.conf.all.disable_ipv6=0
 
   node-red:
-    image: ghcr.io/shankari/everest-demo/nodered:0.0.1
+    image: ghcr.io/everest/everest-demo/nodered:0.0.2
```
Comment (shankari) [inline docker-compose.iso15118-dc.yml:22]:
```suggestion
    image: ghcr.io/everest/everest-demo/nodered:build_pkg_fix_dc_config
```

**Review by shankari — CHANGES_REQUESTED** [review]
Since there are no packages in everest, and there will not be any until the build process is finalized
The only currently built packages are here:
https://github.com/orgs/US-JOET/packages

**Comment by shankari** [comment]
@drmrd are you sure that this works as checked in? I tried testing it and I wasn't able to launch the node-red server or the UI
Screen Shot 2023-11-20 at 9 52 44 PM

### Logs from the launch

- Checking out this branch

```
$ git checkout --track origin/add-15118-dash-2-logs-to-dc-demo
```

- Launching the docker compose

```
$ docker compose -f docker-compose.iso15118-dc.yml up
[+] Building 0.0s (0/0)                                                             docker:desktop-linux
[+] Running 3/0
 ✔ Container everest-demo-mqtt-server-1  Created                                                    0.0s
 ✔ Container everest-demo-node-red-1     Created                                                    0.0s
 ✔ Container everest-demo-manager-1      Created                                                    0.0s
Attaching to everest-demo-manager-1, everest-demo-mqtt-server-1, everest-demo-node-red-1
```

- Everest has started up

```
everest-demo-manager-1      | 2023-11-21 05:49:34.890996 [INFO] manager          :: 🚙🚙🚙 All modules are initialized. EVerest up and running [2087ms] 🚙🚙🚙
everest-demo-manager-1      | 2023-11-21 05:49:37.582542 [INFO] evse_manager:Ev  :: 🌀🌀🌀 Ready to start charging 🌀🌀🌀
```

- But we get MQTT errrors while trying to connect node-red

```
everest-demo-mqtt-server-1  | 1700545789: New connection from 192.168.65.1:38789 on port 1883.
everest-demo-mqtt-server-1  | 1700545789: Client <unknown> disconnected due to protocol error.
everest-demo-mqtt-server-1  | 1700545789: New connection from 192.168.65.1:38790 on port 1883.
everest-demo-mqtt-server-1  | 1700545789: Client <unknown> disconnected due to protocol error.
everest-demo-mqtt-server-1  | 1700545789: New connection from 192.168.65.1:38791 on port 1883.
everest-demo-mqtt-server-1  | 1700545789: Client <unknown> disconnected due to protocol error.
```

**Comment by couryrr-afs** [comment]
@shankari I think you have a typo in the browser. You will want to access `localhost:1880/ui` to access the `node-red` ui.

**Comment by shankari** [comment]
Ooops! This is why I shouldn't test at 10pm.
This does launch now, but it doesn't include the new logging functionality, since the image has not been updated.
Since this is a common theme, and @drmrd is out, I have created a new branch (`build_pkg_task_1_features`) - you can push all these changes to that branch as well to build an image that includes them, and then update all these PRs to use the image tag.

**Comment by couryrr-afs** [comment]
@shankari if I am reading this correctly the creation of images looks to be postponed until the PR is merged. This is so that there is not functionality within the running container that relies on the code in the PR. To test a PR's changes locally you can build that image yourself when pulling down the code. I am leaning toward adding a `latest` tag to the workflow that would pin the latest PR that is merged. Then in all of the `docker-compose.yml` we can just reference `:latest` for now. Then we do not need to go through an extra branch.

**Comment by shankari** [comment]
@couryrr-afs as you can see from the action and the issue, the current flow is:
- you push to `'build_pkg_**'` (either through PR or direct push)
- images are built and uploaded automatically
- you change the docker-compose to use the new image and re-push (either through PR or direct push)

**Comment by drmrd** [comment]
Closing this PR in favor of US-JOET/everest-demo#3 to accommodate existing Docker build process.

**Comment by drmrd** [comment]
Re-opening for the time being, with the intent being to finish a review after we update the CI in #10.

**Comment by shankari** [comment]
@drmrd if you can resolve the merge conflict and bump up the version, this one has been approved and is good to go.

**Comment by drmrd** [comment]
@shankari: I believe this PR is updated and ready to go.

**Review by shankari — APPROVED** [review]

**Pull request merged**